### ISSUE TITLE
Maybe add this as an installation method

### ISSUE THREAD

**Issue by igrmk** [ogham/dog#30]
You can install dog using a command
```
cargo install --git https://github.com/ogham/dog dog
```

It is simpler than cloning a repository, and it already works. Maybe add this as an installation method in the docs?
However I would prefer to simply write
```
cargo install dog
```
if it is possible

**Comment by ogham** [comment]
I'd like to allow `cargo install dog`, but I'm not sure whether it's possible, because the crate is made up of three sub-crates (`dns`, `dns-transport`, and `dog`). Cargo wants them all to be public.

**Comment by mgedmin** [comment]
https://crates.io/crates/dog is now taken by something unrelated (A Datadog API Client for Rust.), so you've missed the opportunity :(

**Comment by horseinthesky** [comment]
@ogham It would still be awesome to install it via cargo as `dns-dog` or something else, please.

**Comment by Siilwyn** [comment]
@horseinthesky alternatively try out `bore`: https://crates.io/crates/bore

**Comment by horseinthesky** [comment]
> @horseinthesky alternatively try out `bore`: [crates.io/crates/bore](https://crates.io/crates/bore)

Thanks but it is messy and ugly as `dig` :P

I like `dog` for efficiency, simplicity, and readability.

**Comment by Walther** [comment]
Kindest bump - would it be possible to re-evaluate the possibility for this?

Example motivation: using [cargo-update](https://github.com/nabijaczleweli/cargo-update) either manually or as part of using [topgrade](https://github.com/r-darwish/topgrade) can be a really nice way to install and keep rusty software up to date on systems where the native package manager repositories might be less up to date - e.g. most non-rolling-release linux distros.

Are the issues with cargo workspaces or something else?

**Comment by im-n1** [comment]
This line

```sh
$ cargo install --git https://github.com/ogham/dog dog
```

does really needs to be added to readme.

**Comment by ActuallyHappening** [comment]
Publishing this to crates.io would be great, `cargo install dog` is what I try every time to remember this project is one of the few I need to use git for

**Comment by theAkito** [comment]
> Publishing this to crates.io would be great, `cargo install dog` is what I try every time to remember this project is one of the few I need to use git for

Yeah, but this project did not receive any commits since 4 years, so it won't happen and is not up to date anyway, therefore security might be compromised more than in an up to date project.

**Comment by ActuallyHappening** [comment]
true true, I'm now using `cargo install dns-doge`

**Comment by theAkito** [comment]
> true true, I'm now using `cargo install dns-doge`

Didn't find anything on that.

Only found [this](https://github.com/Dj-Codeman/dog_community).
